spacepy/pybats/__init__.py

```diff
@@ -877,49 +1093,249 @@ class IdlFile(PbData):
     data contained within the returned object.
 
     This class serves as a parent class to SWMF component-specific derivative
-    classes that do more preprocessing of the data before returning the 
+    classes that do more preprocessing of the data before returning the
     object.  Hence, using this class to read binary files is typically not
     the most efficient way to proceed.  Look for a PyBats sub module that suits
     your specific needs, or use this base object to write your own.
 
-    A note on byte-swapping: PyBats assumes little endian byte ordering because
+    Multi-Frame Files
+    -----------------
+
+    Typically, Idl-formatted data has a single *frame*, or a single snapshot
+    worth of data (a `*.out` file).  However, it is possible to (externally)
+    combine many of these files together such that a time series of data frames
+    are contained within a single file (`*.outs` files). This class can read
+    these files, but only one data frame can be made available at a time.  This
+    prevents very large memory requirements.
+
+    These files are opened and handled similarly to regular IDL-formatted
+    files  with some important differences.  Upon instantiation, the user
+    may select which data frame to open with the *iframe* kwarg.  The default
+    action is to open the first frame.  The user may learn more about the
+    number of frames within the file and their associated epoch/iteration
+    information by examining the top level *attrs* (see below.)
+    The user may switch to any arbitrary frame using the `switch_frame(iframe)`
```

Comment:
Just to note that I really would like to have the ability to iterate over frames. I'm happy to merge without, since there are additional updates coming and it could be argued to be an extra feature request...
But iteration would be super-handy, as would a simple `.next_frame()` or something to help navigation (maybe also `.prev_frame()`). Currently getting the next frame involves looking up the `iframe` in the attrs and passing that through to `switch_frame` which feels a bit messy. Again, noting as a _please put something like this in the upcoming set of revisions_.

Reply:
These are all excellent suggestions.  Only hitch is ensuring we can still iterate over variables (a la dictionary keys/datamodel keys), which is used reasonably often enough.  I'm sure we can work around that, however.  The next and prev frame suggestions are great and are on my to-do list.